google/cloud/spanner_v1/transaction.py

```diff
@@ -368,21 +412,50 @@ def batch_update(self, statements, request_options=None):
         }
         request = ExecuteBatchDmlRequest(
             session=self._session.name,
-            transaction=transaction,
             statements=parsed,
             seqno=seqno,
             request_options=request_options,
         )
-        with trace_call("CloudSpanner.DMLTransaction", self._session, trace_attributes):
-            response = api.execute_batch_dml(request=request, metadata=metadata)
+
+        method = functools.partial(
+            api.execute_batch_dml,
+            request=request,
+            metadata=metadata,
+        )
+
+        if self._transaction_id is None:
```

Comment:
Can you add a comment explaining why lock is added for future references

Reply:
Done